src/core.c/CompUnit/PrecompilationStore/File.pm6

```diff
@@ -368,6 +370,10 @@ class CompUnit::PrecompilationStore::File
          }
          $compiler-dir.rmdir;
     }
+
+    method !tmp-extension(--> Str:D) {
+        "." ~ now.Num ~ "." ~ nqp::getpid() ~ "." ~ nqp::objectid($*THREAD) ~ ".tmp";
```

Comment:
Wouldn't it be easier to just use a random number like `(^2**128).pick.base(36)`?

Reply:
Random doesn't mean unique. The above combination could happen to be non-unique in the only case when within one thread the method is called twice so fast that `now` is not changed. Well, maybe in a far away future... ;)